### ISSUE TITLE
How to set progressive training in traing the VAR?

### ISSUE THREAD

**Issue by HaozheZhao** [FoundationVision/VAR#29]
Great job indeed!

Currently, I am attempting to train the VAR by utilizing the script and code you provided. Upon inspection of the training code and args list, it appears that there are three args, namely pg, pg0, and pgwp, designated for progressive training. I am curious about the hyperparameters you configured for your training, as well as the specific model chosen for progressive training.

Furthermore, I am interested in understanding why the VAR class's self.prog_si in var.py is initially set to -1 and remains unchanged throughout. It seems that neither train.py nor trainer.py reset the prog_si attribute of the VAR class.

**Comment by krennic999** [comment]
I am also curious about the progressive training, can you provide more details?

**Comment by krennic999** [comment]
The prog_si of VAR is modified in train_step in trainer.py, and the default pg seems to be 0 in the whole training loop, pgwp seems to be the setup in warmup during training

**Comment by krennic999** [comment]
Yes, I tend to think so (but not very sure)

**Comment by krennic999** [comment]
hoping the author to give a more concise explanation

**Comment by JamesHujy** [comment]
interested in progressive training as well. looking forward to the explanation

**Comment by keyu-tian** [comment]
Hi @HaozheZhao @krennic999 @JamesHujy, we only use this progressive training when the computation cost is limited, e.g., training d36-s on 512x512 images, by runing with ` --pg=0.7--pg0=4`. For explanation on these hyperparameters, see utils/args.py. This configuration means doing progressive training in the 0%-70% training phase, and full-scale training in the 70%-100% phase. `pgwp` is automatically set in utils/args.py by `args.pgwp = args.ep * 1/300`.
